internal/asm/c128/dscalinc_amd64.s

```diff
@@ -0,0 +1,67 @@
+// Copyright ©2017 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//+build !noasm,!appengine
+
+#include "textflag.h"
+
+#define SRC SI
+#define DST SI
+#define LEN CX
+#define TAIL BX
+#define INC R9
+#define INC3 R10
+#define ALPHA X0
+#define ALPHA_2 X1
+
+// func DscalInc(alpha float64, x []complex128, n, inc int)
```

Comment:
The exact details will be in the initial commits to the asm package in gonum/internal, but the rough picture is that `uintptr` and `int` are not necessarily the same size (amd64p32) and it is easier to pick this failure in the Go code than in the asm. For the common case though, there is no casting since the size is the same, it's just a type annotation.